### PR TITLE
MG-257: model_details.jax_id should preserve leading 0s

### PR DESCRIPTION
**Pull request by beatrizsaldana**
## Problem

JAX IDs in the model_details transformation were being stored as integers, which caused leading zeros to be lost. For example, JAX ID `4807` should be displayed as `004807` to maintain the proper 6-digit format that JAX uses for their mouse strain identifiers.

## Solution

Added logic in the `transform_model_details` function to preserve leading zeros in JAX IDs by converting them to strings with proper 6-digit zero-padding. The solution:

1. **Added JAX ID formatting logic** in `src/agoradatatools/etl/transform/model_details.py` (lines 216-220):
   ```python
   # Ensure jax_id preserves leading zeros by converting to string with proper formatting
   if "jax_id" in model_info_df.columns:
       model_info_df["jax_id"] = model_info_df["jax_id"].apply(
           lambda x: str(x).zfill(6) if pd.notna(x) and str(x).strip() != "" else x
       )
   ```

2. **Updated test assets** to reflect the new string format with leading zeros:
   - Input files continue to use integer format (e.g., `4807`, `5491`)
   - Output files now show properly formatted strings (e.g., `"004807"`, `"005491"`)

The implementation handles edge cases by:
- Only applying formatting when the `jax_id` column exists
- Preserving null/empty values without modification
- Using `zfill(6)` to ensure consistent 6-digit formatting

## Testing

The changes are verified through the existing test suite:

- **Input test files** contain JAX IDs as integers (e.g., `4807`, `5491`, `4808`, `5492`)
- **Output test files** now contain properly formatted strings with leading zeros (e.g., `"004807"`, `"005491"`, `"004808"`, `"005492"`)
- **All existing tests pass** with the updated expected outputs
- **Edge case handling** is tested through existing test scenarios that include missing data and empty values

The transformation maintains backward compatibility while ensuring JAX IDs are displayed in the correct format for downstream applications and user interfaces.

### CODE REVIEW

**Review by JessterB — APPROVED** [review]
LGTM!

File: src/agoradatatools/etl/transform/model_details.py [hunk 1]
```diff
@@ -213,6 +213,12 @@ def transform_model_details(
     model_info_df = datasets["model_info"].fillna("")
     human_transgene_allele_map_df = datasets["human_transgene_allele_map"].fillna("")
 
+    # Ensure jax_id preserves leading zeros by converting to string with proper formatting
```
Comment (thomasyu888) [inline src/agoradatatools/etl/transform/model_details.py:216]: Nit: reading this comment made me think that the input data was as string with leading 0's but really this is a transform to append leading 0s if `jax_id` is not null, is that correct?

Also -  is it ok for `jax_id` to be null?

Comment (jaclynbeck-sage) [reply src/agoradatatools/etl/transform/model_details.py:216]: @thomasyu888 I _think_ so. Last I looked, there was at least one model that was so fresh it hadn't been added to Jax yet so it didn't have a Jax ID.

File: tests/test_assets/model_details/output/model_details_transform_empty_measurements_output.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "contributing_group": "UCI",
     "study_synid": "syn22964685",
     "rrid": "MMRRC_034830-JAX",
-    "jax_id": 4807,
+    "jax_id": "004807",
```
Comment (jaclynbeck-sage) [inline tests/test_assets/model_details/output/model_details_transform_empty_measurements_output.json:11]: Do you have any tests where an input _is_ 6 digits long, just to test that more 0's don't get added?

**Review by jaclynbeck-sage — APPROVED** [review]
Approving as long as there is a test case for a 6-digit-long input. Otherwise this looks great.

File: tests/test_assets/model_details/output/model_details_transform_empty_measurements_output.json (hunk 2 above)
Comment (beatrizsaldana) [reply tests/test_assets/model_details/output/model_details_transform_empty_measurements_output.json:11]: OMG what a good idea! I did not consider that scenario!

Comment (beatrizsaldana) [reply tests/test_assets/model_details/output/model_details_transform_empty_measurements_output.json:11]: Test added!
